broker-core/src/test/java/io/zeebe/broker/workflow/message/MessageStartEventTest.java

```diff
@@ -0,0 +1,299 @@
+        .getFirst();
+
+    // when
+    testClient.publishMessage(MESSAGE_NAME1, "order-123", asMsgPack("foo", "bar"));
+
+    // then
+    final List<Record<WorkflowInstanceRecordValue>> records =
+        RecordingExporter.workflowInstanceRecords().limit(5).collect(Collectors.toList());
+
+    assertThat(records.stream().map(r -> r.getMetadata().getIntent()))
+        .containsExactly(
+            WorkflowInstanceIntent.EVENT_OCCURRED, // message
+            WorkflowInstanceIntent.ELEMENT_READY, // workflow instance
+            WorkflowInstanceIntent.ELEMENT_ACTIVATED,
+            WorkflowInstanceIntent.EVENT_TRIGGERING, // start event
+            WorkflowInstanceIntent.EVENT_TRIGGERED);
+
+    assertThat(records.stream().allMatch(r -> r.getValue().getWorkflowKey() == workflowKey))
+        .isTrue();
+
+    assertThat(records.get(3).getValue().getElementId()).isEqualTo(EVENT_ID1);
+    final WorkflowInstanceRecordValue eventTriggeredRecord = records.get(4).getValue();
+    assertThat(eventTriggeredRecord.getPayloadAsMap().get("foo")).isEqualTo("bar");
+  }
+
+  @Test
+  public void shouldApplyOutputMappingsOfMessageStartEvent() {
+    // given
+    final ExecuteCommandResponse response =
+        testClient.deployWithResponse(createWorkflowWithMessageStartEventOutputMapping());
+
+    final long workflowKey = (long) getDeployedWorkflow(response, 0).get("workflowKey");
+
+    // wait until subscription is opened
+    RecordingExporter.messageStartEventSubscriptionRecords(
+            MessageStartEventSubscriptionIntent.OPENED)
+        .limit(1)
+        .getFirst();
+
+    // when
+    testClient.publishMessage(MESSAGE_NAME1, "order-123", asMsgPack("foo", "bar"));
+
+    // then
+    final Record<WorkflowInstanceRecordValue> record =
+        RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.EVENT_TRIGGERED)
+            .limit(1)
+            .getFirst();
+
+    assertThat(record.getValue().getPayloadAsMap().get("mappedfoo")).isEqualTo("bar");
+  }
+
+  @Test
+  public void shouldCreateInstancesForMultipleMessagesOfSameName() {
+    // given
+    final ExecuteCommandResponse response =
+        testClient.deployWithResponse(createWorkflowWithOneMessageStartEvent());
+
+    final long workflowKey = (long) getDeployedWorkflow(response, 0).get("workflowKey");
+
+    // wait until subscription is opened
+    RecordingExporter.messageStartEventSubscriptionRecords(
+            MessageStartEventSubscriptionIntent.OPENED)
+        .limit(1)
+        .getFirst();
+
+    // when
+    testClient.publishMessage(MESSAGE_NAME1, "order-123", asMsgPack("foo", "bar"));
+    testClient.publishMessage(MESSAGE_NAME1, "order-124", asMsgPack("foo", "bar"));
+
+    // then
+
+    // check if two instances are created
+    final List<Record<WorkflowInstanceRecordValue>> records =
+        RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.ELEMENT_READY)
+            .limit(2)
+            .collect(Collectors.toList());
```

Comment:
Just as a side note, there exists a shortcut called `asList()` for this collect method